### PR TITLE
Ignore permission errors when saving to cache

### PR DESCRIPTION
**Pull request by davidhalter**
This might happen when a user doesn't have full access to his home directory.
Fixes davidhalter/jedi#1615

### CODE REVIEW

File: test/test_cache.py [hunk 1]
```diff
@@ -159,16 +162,29 @@ def test_inactive_cache(tmpdir, isolated_parso_cache):
...
+def test_permission_error(monkeypatch, recwarn):
+    def save(*args, **kwargs):
+        was_called[0] = True  # Python 2... Use nonlocal instead
+        raise PermissionError
+
+    was_called = [False]
+
+    monkeypatch.setattr(cache, '_save_to_file_system', save)
+    parse(path=__file__, cache=True, diff_cache=True)
+    assert was_called[0]
```
Comment (isidentical) [inline test/test_cache.py:190]: Would you also add something to verify the warning has been raised. Something like `pytest.warns(Warning)` or `len(recwarn) > 0`

File: parso/_compatibility.py [hunk 2]
```diff
@@ -50,6 +50,12 @@ def u(string):
 except NameError:
     # Python 2.7 (both IOError + OSError)
     FileNotFoundError = EnvironmentError
+try:
+    # Python 2.7
+    PermissionError = PermissionError
+except NameError:
+    # Python 3.3+
```
Comment (isidentical) [inline parso/_compatibility.py:57]: I find this comments a bit confusing, shouldn't this be `Python 2.7` and the upper one `Python 3.3+`?

File: test/test_cache.py (hunk 1 above)
Comment (davidhalter) [reply test/test_cache.py:190]: Very good point. I even had `recwarn` in the function signature, but then forgot to implement it. Thanks for mentioning it!

File: parso/_compatibility.py (hunk 2 above)
Comment (davidhalter) [reply parso/_compatibility.py:57]: Yeah it's definitely wrong. No idea how I messed that up :)

**Pull request merged**